src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasCountrySlicer.java

```diff
@@ -0,0 +1,502 @@
+        }
+
+        // Slice the JTS Geometry
+        try
+        {
+            return this.countryBoundaryMap.slice(line.getIdentifier(), geometry);
+        }
+        catch (final TopologyException e)
+        {
+            logger.error("Topology Exception when slicing {}", line.getIdentifier(), e);
+            return Collections.emptyList();
+        }
+    }
+
+    /**
+     * Assigns {@link ISOCountryTag} and {@link SyntheticNearestNeighborCountryCodeTag} values for a
+     * given {@link Geometry}, which is a {@link Line} since we don't have other non-Point
+     * geometries in the raw Atlas.
+     *
+     * @param geometry
+     *            The {@link Geoemtry} to create the tags for
+     * @return the resulting tags
+     */
+    private Map<String, String> createLineTags(final Geometry geometry)
+    {
+        final Map<String, String> tags = new HashMap<>();
+        final String countryCode = CountryBoundaryMap.getGeometryProperty(geometry,
+                ISOCountryTag.KEY);
+        tags.put(ISOCountryTag.KEY, countryCode);
+        final String usingNearestNeighbor = CountryBoundaryMap.getGeometryProperty(geometry,
+                SyntheticNearestNeighborCountryCodeTag.KEY);
+        if (usingNearestNeighbor != null)
+        {
+            tags.put(SyntheticNearestNeighborCountryCodeTag.KEY, usingNearestNeighbor);
+        }
+        return tags;
+    }
+
+    /**
+     * Creates a new {@link TemporaryPoint} at the given {@link Coordinate}. We can assume that any
+     * new Points will be created at the country boundaries, so we can add the
+     * {@link SyntheticBoundaryNodeTag} tag.
+     *
+     * @param coordinate
+     *            The {@link Coordinate} of the new point
+     * @param pointIdentifierFactory
+     *            The {@link CountrySlicingIdentifierFactory} to calculate new point identifier
+     * @param pointTags
+     *            The tags for this new point
+     * @return an {@link Optional} containing the {@link TemporaryPoint}
+     */
+    private Optional<TemporaryPoint> createNewPoint(final Coordinate coordinate,
+            final CountrySlicingIdentifierFactory pointIdentifierFactory,
+            final Map<String, String> pointTags)
+    {
+        if (!pointIdentifierFactory.hasMore())
+        {
+            logger.error(
+                    "Country Slicing exceeded maximum number of supported new points at Coordinate {}",
+                    AbstractIdentifierFactory.IDENTIFIER_SCALE, coordinate);
+            return Optional.empty();
+        }
+        else
+        {
+            // Add the synthetic boundary node tags
+            pointTags.put(SyntheticBoundaryNodeTag.KEY, SyntheticBoundaryNodeTag.YES.toString());
+
+            return Optional.of(new TemporaryPoint(pointIdentifierFactory.nextIdentifier(),
+                    new Location(Latitude.degrees(coordinate.y), Longitude.degrees(coordinate.x)),
```

Comment:
Here you should use the `JtsLocationConverter` to avoid code duplication.

Reply:
Good call!